src/rgw/rgw_orphan.cc

```diff
@@ -277,9 +277,28 @@ int RGWOrphanSearch::build_all_oids_index()
     string oid = i->get_oid();
     string locator = i->get_locator();
 
-    string name = oid;
-    if (locator.size())
-      name += " (@" + locator + ")";  
+    ssize_t pos = oid.find('_');
+    if (pos < 0) {
+      cout << "unidentified oid: " << oid << ", skipping" << std::endl;
+      /* what is this object, oids should be in the format of <bucket marker>_<obj>,
+       * skip this entry
+       */
+      continue;
+    }
+    string stripped_oid = oid.substr(pos + 1);
```

Comment:
@cbodley the "<bucket>_" is fixed for any oid, whether it has a namespace or whether it doesn't have it. The next char after it is the one that defines whether there's going to be a namespace or not. So an oid that has a namespace will have two underscores, not one.